### ISSUE TITLE
Re-render options in OptionList

### ISSUE THREAD

**Issue by willmcgugan** [Textualize/textual#3304]
The OptionList will render all the options, and then apply styles to them post render to indicate a highlight.

The problem with this approach is that alpha / dim etc are not taken in to account. The result is that text with alpha, or the dim style, don't apply to the new highlight background.

I think we can re-render highlighted items so that the alpha applies.

**Comment by willmcgugan** [comment]
This will require a fairly substantial refactor.

Essentially the approach of pre-rendering content, and then applying styles to get modifiers like select / hover, isn't going to take alpha in to account.

We need to re-render options with different component classes.

This may be one for @davep since he built the original.

**Comment by rodrigogiraoserrao** [comment]
I can do it, no problem.
But if everyone thinks it's better, I'll yield to Dave.

**Comment by rodrigogiraoserrao** [comment]
TL;DR: This will _likely_ be fixed if [Textual#3342](https://github.com/Textualize/textual/issues/3342) is fixed first.

**Comment by willmcgugan** [comment]
Recently fixed
